components/browser/engine-gecko-beta/README.md

```diff
@@ -12,6 +14,46 @@ Use gradle to download the library from JCenter:
 implementation "org.mozilla.components:engine-gecko-beta:{latest-version}
 ```
 
+### Initializing
+
+It is recommended to create only one `GeckoEngine` instance per app. To create a `GeckoEngine` a `GeckoRuntime` and optionally `DefaultSettings` are needed.
+
+```Kotlin
+// Create default settings (optional) and enable tracking protection for all future sessions.
+val defaultSettings = DefaultSettings().apply {
+    trackingProtectionPolicy = EngineSession.TrackingProtectionPolicy.all()
+}
+
+// Create and initialize a Gecko runtime with the default settings.
+val runtime = GeckoRuntime.getDefault(applicationContext)
+
+// Create an engine instance to be used by other components.
+val engine = GeckoEngine(runtime, defaultSettings)
```

Comment:
Maybe in the future we could encapsulated the creation of the  **GeckoEngine** using the [builder pattern](https://en.wikipedia.org/wiki/Builder_pattern), providing some smart defaults values to easy the creating of GeckoEngine, this way the user of the API can use **GeckoEngine** right away and if want something custom just can mix and match.

Reply:
I feel like Kotlin's ability to have default values in constructor parameters makes the builder pattern most obsolete. We could definitely add `GeckoRuntime.getDefault(applicationContext)` as a default parameter here! I'll file that separately.

Reply:
👍